Microsoft.NET.Build.Containers/CreateNewImage.cs

```diff
@@ -122,7 +130,17 @@ public override bool Execute()
         {
             image.Label(label.ItemSpec, label.GetMetadata("Value"));
         }
-        
+
+        foreach (var port in ExposedPorts)
+        {
+            if (int.TryParse(port.ItemSpec, out int portNumber)
```

Comment:
Shouldn't tryparseport happen here? We can log a warning or something if it didn't succeed, and `Image` can assume what it was given would work (or we do the validation both places). I feel like customers should get some explicit message from us if something goes wrong here though. Thoughts?

Reply:
I agree on all counts, but the problem for TryParsePort is that they want the whole string. I suppose we could concat the ItemSpec and Type, if it exists?

Reply:
Another problem with TryParsePort - it's parsing what should be a machine-written value in the container config. This is validating user data, so we are probably ok to not try to reuse that too much and instead be more specific about the user errors here.

Reply:
I, uh, _may_ have done some refactoring here so that both port-parsing points use a shared parsing logic.